### PR TITLE
Fix #5090: Added Local TableType

### LINKED ISSUE #5090: Add "Local" as enum for Redshift table type
Can we add "Local" as enum for table type, since this is standard in Redshift?
https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_AS.html

https://github.com/open-metadata/OpenMetadata/blob/9a3fa2e532ecbdf0c83d0ebce30788ce511fe965/openmetadata-core/src/main/resources/json/schema/entity/data/table.json#L16

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes :

Fix #5090: Added Local TableType

#
### Type of change :

- [x] Bug fix

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ingestion

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/redshift.py [hunk 1]
```diff
@@ -460,6 +468,36 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
...
+        tabel_fqn = fqn.build(
+            self.metadata,
+            Table,
+            database_name=self.context.database.name.__root__,
+            schema_name=schema_name,
+            table_name=table_name,
+            service_name=self.config.serviceName,
+        )
+        if not self.table_type_map:
+            table_type_query = "SELECT database_name,schema_name, table_name,table_type FROM svv_all_tables"
```
Comment (ulixius9) [inline ingestion/src/metadata/ingestion/source/database/redshift.py:484]: reference: https://docs.aws.amazon.com/redshift/latest/dg/r_SVV_ALL_TABLES.html

File: ingestion/src/metadata/ingestion/source/database/redshift.py [hunk 2]
```diff
@@ -460,6 +468,36 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
...
+            self.table_type_map = {
+                fqn.build(
+                    self.metadata,
+                    Table,
+                    database_name=row[0],
+                    schema_name=row[1],
+                    table_name=row[2],
+                    service_name=self.config.serviceName,
+                ): row[3]
+                for row in result
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/database/redshift.py:495]: From what I understand (and testing locally) `row` will be a `sqlalchemy.engine.row.LegacyRow` type. We sh🙂ould be able to be explicit when fetching the value of the row (e.g. `row.table_name`, etc.) vs accessing the value via an index.

File: ingestion/src/metadata/ingestion/source/database/redshift.py (hunk 1 above)
Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/database/redshift.py:484]: Thanks @ulixius9, clean approach and we handle external tables properly, which we were missing.

Initially I was thinking about preparing this `get_table_type` method for those cases where we have a static answer, such as Athena, where by def tables are external. However, if we do need to run a query to obtain the proper values, that approach does not look valid anymore. Note that with this solution we will be running a query for each table.

Storing the tables and types as we do with `tags` might not be the best approach either, as we might have a bunch of tables. Instead, shall we update:

```
self.inspector.get_table_names(schema_name):
```

to return the table type as well? In those dialects where this does not exist or it is not necessary, we can always update the dialect definition 🤔 

WDYT?

File: ingestion/src/metadata/ingestion/source/database/redshift.py [hunk 3]
```diff
@@ -441,6 +451,7 @@ class RedshiftSource(CommonDbSourceService):
 
     def __init__(self, config, metadata_config):
         super().__init__(config, metadata_config)
+        self.table_type_map = {}
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/database/redshift.py]: This seems like it is not used

**Review by pmbrull — APPROVED** [review]

**Pull request merged**